packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -325,6 +335,24 @@ class TFContracts extends Contracts {
     return ids;
   }
 
+  async batchUnlockContracts(ids: number[]) {
+    const extrinsics: ExtrinsicResult<Contract>[] = [];
+    for (const id of ids) {
+      extrinsics.push(await this.unlock(id));
+    }
+    return this.client.applyAllExtrinsics(extrinsics);
+  }
+  async unlockMyContracts(graphqlURL: string) {
+    const contracts = await this.listMyContracts({
+      stateList: [ContractStates.GracePeriod],
+      graphqlURL,
+    });
```

Comment:
Can we filter contracts with zero amounts locked to reduce network load and user fees?

Reply:
the idea is we are listing contracts using gql and it doesn't return any billing info.
so we have to make another call to get billing info. this we increase the network load, what do you think?
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/41be566c-5c6b-4275-8b14-b426bf9481d8)

Reply:
Querying a value from the chain is generally much less computationally expensive than sending an extrinsic. When querying the storage, it's like retrieving/reading a value from the db of any of the RPC nodes, which is fast and requires no propagation across the network, multiple executions on all validators, or inclusion in a new block on the chain. On the other hand, sending extrinsic would require all of these, thus necessitating the sender to pay fees for the consumed resources.